website/cue/reference/components/sources/internal_metrics.cue

```diff
@@ -46,6 +46,15 @@ components: sources: internal_metrics: {
 	}
 
 	configuration: {
+		namespace: {
+			description: "The namespace prefixed to a metric's name."
```

Comment:
This description is copied from other components, but I'm not sure it's entirely accurate.

When the codec used is "text", the format is output as `<namespace>_<metric_name>`, but when the codec is "json", it's provided as a dedicated `namespace` field.

This description would lead me to believe that the `name` field itself would be modified.

I opted for prior art in this case, but just an FYI.

Reply:
🤔 perhaps specifying between "prometheus" naming (currently described) and an "object based" event?

Reply:
I'd be content with `The namespace of the metric`. The prefixed bit is sink specific (e.g. the `prometheus_exporter` sink).

Reply:
I saw it with both prometheus and influxdb, and assumed this behaviour was global.

I see that the [data_model describes it accurately](https://github.com/timberio/vector/blob/5dd7ffc4febbebdcfc0519458d3f29b6c01b90b5/website/cue/reference/data_model/schema.cue#L254-L262) already:

```
The metric namespace. Depending on the service, this will prepend the name or use native namespacing facilities.
```

Will update to "The namespace of the metric" 👍🏻